monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/AvailablePlugins.tsx

```diff
@@ -0,0 +1,125 @@
+  const {refreshAvailablePlugins} = useContext(PluginsContext);
+  const {refreshInstalledPlugins} = useContext(PluginsContext);
+  const [displayedPlugins, setDisplayedPlugins] = useState([]);
+
+  const [successfullyInstalledPluginsIds, setSuccessfullyInstalledPluginsIds] = useState([]);
+  const [pluginsInInstallationProcess, setPluginsInInstallationProcess] = useState([]);
+  const authComponent = new AuthComponent({});
+
+  useEffect(() => {
+    let installedPluginsIds = installedPluginsToArray(installedPlugins).map(generatePluginId);
+    const installedFilter = (plugin) => !installedPluginsIds.includes(generatePluginId(plugin));
+    let shownPlugins = pluginIndexToArray(availablePlugins).filter(installedFilter)
+    setDisplayedPlugins(shownPlugins);
+  }, [installedPlugins, availablePlugins]);
+
+  const onRefreshCallback = () => {
+    setSuccessfullyInstalledPluginsIds([]);
+  }
+
+  const onInstallClick = (pluginId, pluginName, pluginType, pluginVersion) => {
+    console.log('installing plugin: ', pluginName)
+
+    setPluginsInInstallationProcess((prevState) => {
+      return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+    });
+
+    authComponent.authFetch('/api/install-agent-plugin', {method: 'PUT', headers: {'Content-Type': 'application/json'}, body: JSON.stringify({plugin_type: pluginType, name: pluginName, version: pluginVersion})}, true).then(() => {
+      setSuccessfullyInstalledPluginsIds((prevState) => {
+        return shallowAdditionOfUniqueValueToArray(prevState, pluginId);
+      });
+      refreshInstalledPlugins();
+    }).catch(() => {
+      console.log('error installing plugin');
+    }).finally(() => {
+      setPluginsInInstallationProcess((prevState => {
+        return shallowRemovalOfUniqueValueFromArray(prevState, pluginId);
+      }));
+    });
+  };
+
+  const getRowActions = (row) => {
+    const pluginId = row.id;
+    if (pluginsInInstallationProcess.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={nanoid()}
+          icon={<DownloadingIcon/>}
+          label="Downloading"
+          className="textPrimary"
+          color="inherit"
+        />
+      ]
+    }
+
+    if (successfullyInstalledPluginsIds.includes(pluginId)) {
+      return [
+        <GridActionsCellItem
+          key={nanoid()}
+          icon={<DownloadDoneIcon/>}
+          label="Download Done"
+          className="textPrimary"
+          color="inherit"
+        />
+      ]
+    }
+
+    const pluginName = row.name;
+    const pluginType = row.type;
+    const pluginVersion = row.version;
+    return [
+      <GridActionsCellItem
+        key={nanoid()}
+        icon={<FileDownloadIcon/>}
+        label="Download"
+        className="textPrimary"
+        onClick={() => onInstallClick(pluginId, pluginName, pluginType, pluginVersion)}
+        color="inherit"
+      />
+    ];
+  }
+
+  return (
+    <Box>
+      <Row className='grid-tools'>
```

Comment:
Don't use Bootstrap's col/row/button, but use the MUI ones.
Using flex properties on the tools wrapper might do the job.
In case you need a custom stylesheet- I would suggest creating a new module for it.